spring-cloud-stream-binder-kafka-streams/src/main/java/org/springframework/cloud/stream/binder/kafka/streams/KafkaStreamsStreamListenerSetupMethodOrchestrator.java

```diff
@@ -288,8 +296,48 @@ else if (parameterType.isAssignableFrom(KTable.class)) {
 						.withValueSerde(v));
 	}
 
-	private KStream<?, ?> getkStream(String inboundName, BindingProperties bindingProperties, StreamsBuilder streamsBuilder,
-									Serde<?> keySerde, Serde<?> valueSerde) {
+	private StoreBuilder buildStateStore(KafkaStreamsStateStoreProperties spec) throws Exception {
+		try {
+			Serde<?> keySerde = this.keyValueSerdeResolver.getStateStoreKeySerde(spec.getKeySerdeString());
+			Serde<?> valueSerde = this.keyValueSerdeResolver.getStateStoreValueSerde(spec.getValueSerdeString());
+			StoreBuilder builder = null;
+			switch (spec.getType()) {
+				case KEYVALUE:
+					builder = Stores.keyValueStoreBuilder(Stores.persistentKeyValueStore(spec.getName()), keySerde, valueSerde);
+					break;
+				case WINDOW:
+					builder = Stores.windowStoreBuilder(Stores.persistentWindowStore(spec.getName(), spec.getRetention(), 3, spec.getLength(), false),
+							keySerde,
+							valueSerde);
+					break;
+				case SESSION:
+					builder = Stores.sessionStoreBuilder(Stores.persistentSessionStore(spec.getName(), spec.getRetention()), keySerde, valueSerde);
+					break;
+				default:
+					throw new UnsupportedOperationException("state store type (" + spec.getType() + ") is not supported!");
+			}
+			if (spec.isCacheEnabled())
+				builder = builder.withCachingEnabled();
```

Comment:
To comply with spring styles please include open/close `{}` braces around IFs and other similar blocks of code.